src/scales/logarithmic.js

```diff
@@ -0,0 +1,121 @@
+import {BaseScale} from './base';
+import {utils} from '../utils/utils';
+/* jshint ignore:start */
+import {default as _} from 'underscore';
+import {default as d3} from 'd3';
+/* jshint ignore:end */
+
+export class LogarithmicScale extends BaseScale {
+
+    constructor(xSource, scaleConfig) {
+
+        super(xSource, scaleConfig);
+
+        var isNum = ((num) => (!isNaN(num) && _.isNumber(num)));
+
+        var props = this.scaleConfig;
+        var vars = d3.extent(this.vars);
+
+        var min = isNum(props.min) ? props.min : vars[0];
+        var max = isNum(props.max) ? props.max : vars[1];
+
+        vars = [
+            Math.min(...[min, vars[0]].filter(isNum)),
+            Math.max(...[max, vars[1]].filter(isNum))
+        ];
+
+        if (props.nice) {
+            if (crossesZero(vars)) {
+                vars = utils.niceZeroBased(vars);
+            } else {
+                vars = utils.niceLog10(vars);
+            }
+        }
+
+        this.vars = vars;
+
+        this.addField('scaleType', 'logarithmic')
+            .addField('discrete', false);
+    }
+
+    isInDomain(x) {
+        var domain = this.domain();
+        var min = domain[0];
+        var max = domain[domain.length - 1];
+        return (!isNaN(min) && !isNaN(max) && (x <= max) && (x >= min));
+    }
+
+    create(interval) {
+
+        var vars = this.vars;
+
+        var d3Scale;
+
+        if (crossesZero(vars)) {
+            /*eslint-disable */
+            console.warn(
```

Comment:
Let's throw exception and handle it somewhere on high level where we have access to custom taucharts logger and a able to replace invalid log scale to linear (and make it optionally).